python/ray/data/tests/test_map.py

```diff
@@ -330,6 +331,175 @@ def map_generator(item: dict) -> Iterator[int]:
+    [
+        pytest.param(
+            pd.DataFrame(
+                {
+                    "id": [1, 2, 3],
+                    "timestamp": pd.to_datetime(
+                        [
+                            "2024-01-01 00:00:00.123456789",
+                            "2024-01-02 00:00:00.987654321",
+                            "2024-01-03 00:00:00.111222333",
+                        ]
+                    ),
+                    "value": [10.123456789, 20.987654321, 30.111222333],
+                }
+            ),
+            pd.DataFrame(
+                {
+                    "id": [1, 2, 3],
+                    "timestamp": pd.to_datetime(
+                        [
+                            "2024-01-01 00:00:00.123456790",
+                            "2024-01-02 00:00:00.987654322",
+                            "2024-01-03 00:00:00.111222334",
+                        ]
+                    ),
+                    "value": [10.123456789, 20.987654321, 30.111222333],
+                }
+            ),
+            id="nanoseconds_increment",
+        )
+    ],
+)
+def test_map_timestamp_nanosecs(df, expected_df, ray_start_regular_shared):
+    ray_data = ray.data.from_pandas(df)
+    result = ray_data.map(process_timestamp_data)
+    # Convert the result back to a Pandas DataFrame
+    processed_df = result.to_pandas()
+    # Convert PyArrow Timestamps to Pandas Timestamps
+    processed_df["timestamp"] = processed_df["timestamp"].apply(
+        lambda x: pd.Timestamp(x.as_py())
+        if isinstance(x, pa.lib.TimestampScalar)
+        else x
+    )
+    # Ensure the dtype is correct
+    processed_df["timestamp"] = processed_df["timestamp"].astype("datetime64[ns]")
+    pd.testing.assert_frame_equal(processed_df, expected_df)
+
+
+@pytest.mark.parametrize(
+    "df, expected_df",
+    [
+        pytest.param(
+            pd.DataFrame(
+                {
+                    "id": [1, 2, 3],
+                    "timestamp": [
+                        np.datetime64("2024-01-01T00:00:00.123456789"),
+                        np.datetime64("2024-01-02T00:00:00.987654321"),
+                        np.datetime64("2024-01-03T00:00:00.111222333"),
+                    ],
+                    "value": [10.123456789, 20.987654321, 30.111222333],
+                }
+            ),
+            pd.DataFrame(
+                {
+                    "id": [1, 2, 3],
+                    "timestamp": pd.to_datetime(
+                        [
+                            "2024-01-01 00:00:00.123456790",
+                            "2024-01-02 00:00:00.987654322",
+                            "2024-01-03 00:00:00.111222334",
+                        ]
+                    ),
+                    "value": [10.123456789, 20.987654321, 30.111222333],
+                }
+            ),
+            id="numpy_datetime64",
+        )
+    ],
+)
+def test_map_numpy_datetime(df, expected_df, ray_start_regular_shared):
```

Comment:
Nit: It seems like `test_map_numpy_datetime` and `test_map_timestamp_nanosecs` have the same body? Can probably move this to be one test with two different sets of parameters.